src/app/clusters/operational-credentials/operational-credentials.cpp

```diff
@@ -33,33 +36,193 @@
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: RemoveFabric"); // TODO: Generate emberAfFabricClusterPrintln
-    EmberAfStatus status = EMBER_ZCL_STATUS_FAILURE;
+    
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    AdminPairingInfo * admin;
+    AdminId adminId;
+    CHIP_ERROR err;
+
+    // Fetch current admin
+    admin = retrieveCurrentAdmin();
+    VerifyOrExit(admin != nullptr, status = EMBER_ZCL_STATUS_SUCCESS); // Admin has already been removed
+
+    // Delete admin
+    adminId = admin->GetAdminId();
+    GetGlobalAdminPairingTable().ReleaseAdminId(adminId);
+    err = AdminPairingInfo::DeleteFromKVS(adminId);
+    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
+
+exit:
+
     emberAfSendImmediateDefaultResponse(status);
     return true;
 }
 
-bool emberAfOperationalCredentialsClusterGetFabricIdCallback(chip::app::Command * commandObj)
+bool emberAfOperationalCredentialsClusterSetFabricCallback(chip::app::Command * commandObj, uint16_t VendorId)
 {
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: GetFabricId");
-    EmberAfStatus status = EMBER_ZCL_STATUS_FAILURE;
-    emberAfSendImmediateDefaultResponse(status);
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: SetFabric with vendorId %d", VendorId);
+
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    EmberStatus sendStatus = EMBER_SUCCESS;
+    CHIP_ERROR err;
+
+    // Fetch current admin
+    AdminPairingInfo * admin = retrieveCurrentAdmin();
+    VerifyOrExit(admin != nullptr, status = EMBER_ZCL_STATUS_FAILURE);
+
+    // Store vendorId
+    admin->SetVendorId(VendorId);
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: vendorId is now set %d", admin->GetVendorId());
+    err = admin->StoreIntoKVS();
+    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
+
+    // Return FabricId
+    emberAfFillExternalBuffer((ZCL_CLUSTER_SPECIFIC_COMMAND | ZCL_FRAME_CONTROL_SERVER_TO_CLIENT), ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID,
+                              ZCL_SET_FABRIC_RESPONSE_COMMAND_ID, "y", emberAfCurrentCommand()->source);
+    sendStatus = emberAfSendResponse();
+
+exit:
+    if (status == EMBER_ZCL_STATUS_FAILURE)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed setFabricVendorId.");
+        emberAfSendImmediateDefaultResponse(status);
+    }
+    if (sendStatus != EMBER_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to send %s response: 0x%x", "set_fabric", sendStatus);
+    }
+
     return true;
 }
 
 bool emberAfOperationalCredentialsClusterUpdateFabricLabelCallback(chip::app::Command * commandObj, uint8_t * Label)
 {
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: UpdateFabricLabel");
-    // Go look at admin table using fabricId + update the label
-    // Then find equivalent value in attribute list and update it.
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: UpdateFabricLabel to %s", (char*) Label);
```

Comment:
`Label` is not a null-terminated char string; it's a Pascal-style string with length in first byte.

So this is not going to do the right thing.  `emberAfPrintString` or the generated cluster-specific wrapper for it is the way to go here.

Reply:
This was  accidental leftover, meant to remove. Actually am not supporting the label yet because there is something wrong with octet_strings in lists implementation.